### PR TITLE
Allow inline images in the email

### LINKED ISSUE #19: Inline/embedded images in html parts
Would be nice to include (somehow) inline images

### PR DESCRIPTION
**Pull request by aericson**
Closes #19 .

### CODE REVIEW

**Comment by aericson** [comment]
@andersonresende this is ready for review.

File: README.rst [hunk 1]
```diff
@@ -62,7 +66,9 @@ Python to send mail::
             # template_suffix="email",
     )
 
-If you would like finer control on sending the email, you can use **get_templated_email**, which will return a django **EmailMessage** object, prepared using the **vanilla_django** backend::
+If you would like finer control on sending the email, you can use **get_templated_email**, which will return a django **EmailMessage** object, prepared using the **vanilla_django** backend:
```
Comment (andersonresende) [inline README.rst]: Only uses the `vanilla_django`, right?

File: README.rst [hunk 2]
```diff
@@ -137,13 +151,57 @@ You can use it with django-templated-email, just follow their instructions in th
...
+First get the image content from a file or a *ImageField*:
+
+.. code-block:: python
+
+    # From a file
+    with open('lena.png', 'rb') as lena:
+      image = lena.read()
+
+    # From a ImageField ('logo' is a ImageField of Company)
+    image = company.logo.read()
```
Comment (andersonresende) [inline README.rst]: I don't know but would be good make the complete example.
`class Company(...):
 ...
`

File: README.rst (hunk 1 above)
Comment (aericson) [inline README.rst]: Yea, removed all other backends. We should probably clean up and remove all the backends logic at some point.

File: templated_email/__init__.py [hunk 3]
```diff
@@ -1,10 +1,10 @@
 from django.conf import settings
 from django.utils.module_loading import import_string
-from templated_email.backends.vanilla_django import TemplateBackend
 
 import six
-import warnings
-warnings.filterwarnings('error', 'django.utils.importlib')
+
+from templated_email.backends.vanilla_django import TemplateBackend
+from templated_email.utils import InlineImage  # noqa
```
Comment (andersonresende) [inline templated_email/__init__.py]: What is that comment : `#noqa`

Comment (aericson) [inline templated_email/__init__.py]: it's for flake8 to ignore it not being used in the file. But we want it there to allow `from templated_email import InlineImage`.

File: templated_email/utils.py [hunk 4]
```diff
@@ -25,3 +28,38 @@ def _get_klass_from_config(config_variable, default):
...
+
+class InlineImage(object):
+
+    def __init__(self, filename, content, subtype=None, domain=None):
+        self.filename = filename
+        self._content = content
+        self.subtype = subtype
+        self.domain = domain
+        self._content_id = None
+
```
Comment (andersonresende) [inline templated_email/utils.py]: Why not call `generate_cid` here?

**Comment by andersonresende** [comment]
@aericson  I made some comments please take a look

**Pull request merged**